platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/DefaultExtraPropertiesExtension.java

```diff
@@ -100,17 +110,22 @@ public void setProperty(String name, @Nullable Object newValue) {
     @Override
     public Map<String, Object> getProperties() {
         // TODO:configuration-cache use a tracking map here
-        HashMap<String, Object> properties = new HashMap<>(storage);
+        if (storage == null) {
+            return gradleProperties;
+        }
+        ImmutableMap.Builder<String, Object> builder = ImmutableMap.builderWithExpectedSize(storage.size());
```

Comment:
💅  Would the expected size of `storage.size + gradleProperties.size` be more accurate?

Reply:
In other words, would it be more common for users to be overwriting Gradle properties or adding extra properties on top? I guess adding on top would be more common, yes, I'll make the change.